File: devops-scm-provider/devops-scm-provider-git/devops-scm-provider-tgit/src/main/java/com/tencent/devops/scm/provider/git/tgit/auth/TGitTokenAuthProviderAdapter.java
package com.tencent.devops.scm.provider.git.tgit.auth;

import com.tencent.devops.scm.api.pojo.auth.AccessTokenScmAuth;
import com.tencent.devops.scm.api.pojo.auth.IScmAuth;
import com.tencent.devops.scm.api.pojo.auth.PersonalAccessTokenScmAuth;
import com.tencent.devops.scm.api.pojo.auth.PrivateTokenScmAuth;
import com.tencent.devops.scm.api.pojo.auth.TokenSshPrivateKeyScmAuth;
import com.tencent.devops.scm.api.pojo.auth.TokenUserPassScmAuth;
import com.tencent.devops.scm.sdk.tgit.auth.TGitAuthProvider;
import com.tencent.devops.scm.sdk.tgit.auth.TGitTokenAuthProvider;

public class TGitTokenAuthProviderAdapter implements TGitAuthProviderAdapter {

    @Override
    public boolean support(IScmAuth auth) {
        return auth instanceof AccessTokenScmAuth
                || auth instanceof PrivateTokenScmAuth
                || auth instanceof PersonalAccessTokenScmAuth
                || auth instanceof TokenUserPassScmAuth
                || auth instanceof TokenSshPrivateKeyScmAuth;
    }

    @Override
    public TGitAuthProvider get(IScmAuth auth) {
        if (auth instanceof AccessTokenScmAuth) {
            return TGitTokenAuthProvider.fromOauthToken(((AccessTokenScmAuth) auth).getAccessToken());
        }
        if (auth instanceof PrivateTokenScmAuth) {
            return TGitTokenAuthProvider.fromPrivateToken(((PrivateTokenScmAuth) auth).getPrivateToken());
        }
        if (auth instanceof PersonalAccessTokenScmAuth) {
            return TGitTokenAuthProvider.fromPersonalAccessToken(
                    ((PersonalAccessTokenScmAuth) auth).getPersonalAccessToken());
        }
        if (auth instanceof TokenUserPassScmAuth) {
            return TGitTokenAuthProvider.fromPrivateToken(((TokenUserPassScmAuth) auth).getToken());
        }
        if (auth instanceof TokenSshPrivateKeyScmAuth) {
            return TGitTokenAuthProvider.fromPrivateToken(((TokenSshPrivateKeyScmAuth) auth).getToken());
        }
        throw new UnsupportedOperationException(String.format("gitAuth(%s) is not support", auth));
    }
}

File: devops-scm-provider/devops-scm-provider-git/devops-scm-provider-tgit/src/main/kotlin/com/tencent/devops/scm/provider/git/tgit/auth/TGitTokenAuthProviderAdapter.kt
package com.tencent.devops.scm.provider.git.tgit.auth

import com.tencent.devops.scm.api.pojo.auth.AccessTokenScmAuth
import com.tencent.devops.scm.api.pojo.auth.IScmAuth
import com.tencent.devops.scm.api.pojo.auth.PersonalAccessTokenScmAuth
import com.tencent.devops.scm.api.pojo.auth.PrivateTokenScmAuth
import com.tencent.devops.scm.api.pojo.auth.TokenSshPrivateKeyScmAuth
import com.tencent.devops.scm.api.pojo.auth.TokenUserPassScmAuth
import com.tencent.devops.scm.sdk.tgit.auth.TGitAuthProvider
import com.tencent.devops.scm.sdk.tgit.auth.TGitTokenAuthProvider

/**
 * TGit Token授权提供者适配器实现类
 */
class TGitTokenAuthProviderAdapter : TGitAuthProviderAdapter {

    /**
     * 判断是否支持该授权类型
     * @param auth 授权信息
     * @return 是否支持
     */
    override fun support(auth: IScmAuth): Boolean {
        return auth is AccessTokenScmAuth ||
                auth is PrivateTokenScmAuth ||
                auth is PersonalAccessTokenScmAuth ||
                auth is TokenUserPassScmAuth ||
                auth is TokenSshPrivateKeyScmAuth
    }

    /**
     * 根据授权信息获取对应的Token授权提供者
     * @param auth 授权信息
     * @return Token授权提供者实例
     * @throws UnsupportedOperationException 当授权类型不支持时抛出
     */
    override fun get(auth: IScmAuth): TGitAuthProvider {
        return when (auth) {
            is AccessTokenScmAuth -> TGitTokenAuthProvider.fromOauthToken(auth.accessToken)
            is PrivateTokenScmAuth -> TGitTokenAuthProvider.fromPrivateToken(auth.privateToken)
            is PersonalAccessTokenScmAuth -> TGitTokenAuthProvider.fromPersonalAccessToken(auth.personalAccessToken)
            is TokenUserPassScmAuth -> TGitTokenAuthProvider.fromPrivateToken(auth.token)
            is TokenSshPrivateKeyScmAuth -> TGitTokenAuthProvider.fromPrivateToken(auth.token)
            else -> throw UnsupportedOperationException("gitAuth($auth) is not support")
        }
    }
}
